### PR TITLE
implimented catch and attack

### PR DESCRIPTION
**Pull request by Ran-Regev-HLS**
I have:
- [ ] deleted all irrelevant comments.
- [ ] deleted all irrelevant logs and debuggers.
- [ ] deleted all sensitive information.
- [ ] made sure my code is generic and reusable for future things or reused existing resources.
- [ ] made sure that my code is readable.
- [ ] checked that the functions and resource names are self-explanatory.
- [ ] tested the code locally and it works as expected.
- [ ] added validations and error handling in relevant cases (external requests, integrations, etc).

### CODE REVIEW

File: src/assets/pages/FightingArenaPage/FightingArenaPage.tsx [hunk 1]
```diff
@@ -27,43 +29,95 @@ import {
...
 
+  function handleAttackClick() {
+    setFightStatus(FightStatus.ATTACKING);
+    const newDefenderHP = handleAttack({ selectedPokemon, opponentPokemon });
+    if (newDefenderHP) {
+      setOpponentPokemon((prev) => ({ ...prev, currentHP: newDefenderHP }));
+    }
+    setTimeout(() => {
+      if (newDefenderHP && newDefenderHP <= 0) {
+        setFightStatus(FightStatus.WON);
```
Comment (kfir-azu) [inline src/assets/pages/FightingArenaPage/FightingArenaPage.tsx:90]: you should run inside this if - handleFightWon function

File: src/assets/pages/FightingArenaPage/FightingArenaPage.tsx [hunk 2]
```diff
@@ -27,43 +29,95 @@ import {
...
 } from "./styles";
-import { convertPokemonsToOptions } from "./util";
+import { convertPokemonsToOptions, handleAttack } from "./utils";
 import { useQuery } from "@tanstack/react-query";
-import { PokemonType } from "../../components/PokemonTable/types";
+import { PokemonInBattle } from "./types";
+import { attemptCatch } from "./CatchMechanism/CatchMechanism";
+import { MAX_CATCH_ATTEMPTS } from "./CatchMechanism/constants";
 
 function FightingArenaPage() {
```
Comment (kfir-azu) [inline src/assets/pages/FightingArenaPage/FightingArenaPage.tsx:38]: how do you set who starts the fight? who's turn is it?
i have seen that every time you press attack you handle this with set timeout and let the opponent attack as well

File: src/assets/pages/FightingArenaPage/FightingArenaPage.tsx [hunk 3]
```diff
@@ -27,43 +29,95 @@ import {
...
+      const newPlayerHP = handleAttack({ opponentPokemon, selectedPokemon });
+      if (newPlayerHP) {
+        setSelectedPokemon((prev) => ({ ...prev, currentHP: newPlayerHP }));
+      }
+      setTimeout(() => {
+        if (newPlayerHP && newPlayerHP <= 0) {
+          setFightStatus(FightStatus.LOST);
+        }
+       
+          setFightStatus((prev) => (prev === FightStatus.ATTACKING ? FightStatus.IN_FIGHT : prev));
```
Comment (kfir-azu) [inline src/assets/pages/FightingArenaPage/FightingArenaPage.tsx:101]: Why do you need fightStatus.Attacking? what happens in this status?
i am not sure i would go to this level with the fightStatus

File: src/assets/pages/FightingArenaPage/FightingArenaPage.tsx [hunk 4]
```diff
@@ -27,43 +29,95 @@ import {
...
+  function handleCatchClick() {
+    setFightStatus(FightStatus.CATCHING);
+    const catchStatus = attemptCatch(opponentPokemon.currentHP);
+    setCatchAttemptsLeft((prev) => prev - 1);
+  
+    setTimeout(() => {
+     if(catchStatus){
+      setFightStatus(FightStatus.CATCHED);
+     }else if (catchAttemptsLeft <= 0) {
+      setFightStatus(FightStatus.LOST);
```
Comment (kfir-azu) [inline src/assets/pages/FightingArenaPage/FightingArenaPage.tsx:114]: I would differentiate this status from lost.
I would add "opponent fled" status as it a bit different from losing the fight when selected pokemon hp === 0

File: src/assets/pages/FightingArenaPage/constans.ts [hunk 5]
```diff
@@ -7,4 +7,16 @@ export const FIGHT_BUTTON_TEXT = "FIGHT"
 
 export const ATTACK_BUTTON_TEXT = "Attack"
 
-export const CATCH_BUTTON_TEXT = "Catch"
\ No newline at end of file
+export const CATCH_BUTTON_TEXT = "Catch"
+
+export enum FightStatus {
```
Comment (kfir-azu) [inline src/assets/pages/FightingArenaPage/constans.ts:12]: Big like on the idea and implementation!
 i have added in fightArena component some suggestions about this

File: src/assets/pages/FightingArenaPage/FightingArenaPage.tsx (hunk 1 above)
Comment (Ran-Regev-HLS) [reply src/assets/pages/FightingArenaPage/FightingArenaPage.tsx:90]: this happens with use effect in later branches (using the fight status)

File: src/assets/pages/FightingArenaPage/FightingArenaPage.tsx (hunk 3 above)
Comment (Ran-Regev-HLS) [reply src/assets/pages/FightingArenaPage/FightingArenaPage.tsx:101]: yeah i see my mistake now - i set it to i start always, this is quite a big change so i will make a branch of its own.

**Review by kfir-azu — APPROVED** [review]

**Pull request closed without merging**